.github/workflows/ci.yml

```diff
@@ -11,12 +11,83 @@ jobs:
-        os: [ubuntu-latest]
-        # Support for 3.6.0 is not available in CI since the Docker image has issues with option management in regard of SASL/OAUTHBEARER.
-        kafka-version: ['3.5.0', '3.7.0', '3.8.0', '3.9.0', '4.0.0']
-    runs-on: ${{matrix.os}}
+        include:
+          - label: Node.js 20.19.5 - Kafka 3.5.0
+            node: 20.19.5
+            confluent-kafka-version: 7.5.0
+            os: ubuntu-24.04
+          - label: Node.js 20.19.5 - Kafka 3.6.0
+            node: 20.19.5
+            confluent-kafka-version: 7.6.0
+            os: ubuntu-24.04
+          - label: Node.js 20.19.5 - Kafka 3.7.0
+            node: 20.19.5
+            confluent-kafka-version: 7.7.0
+            os: ubuntu-24.04
+          - label: Node.js 20.19.5 - Kafka 3.8.0
+            node: 20.19.5
+            confluent-kafka-version: 7.8.0
+            os: ubuntu-24.04
+          - label: Node.js 20.19.5 - Kafka 3.9.0
+            node: 20.19.5
+            confluent-kafka-version: 7.9.0
+            os: ubuntu-24.04
+          - label: Node.js 20.19.5 - Kafka 4.0.0
+            node: 20.19.5
+            confluent-kafka-version: 8.0.0
+            os: ubuntu-24.04
+
+          - label: Node.js 22.19.0 - Kafka 3.5.0
+            node: 22.19.0
+            confluent-kafka-version: 7.5.0
+            os: ubuntu-24.04
+          - label: Node.js 22.19.0 - Kafka 3.6.0
+            node: 22.19.0
+            confluent-kafka-version: 7.6.0
+            os: ubuntu-24.04
+          - label: Node.js 22.19.0 - Kafka 3.7.0
+            node: 22.19.0
+            confluent-kafka-version: 7.7.0
+            os: ubuntu-24.04
+          - label: Node.js 22.19.0 - Kafka 3.8.0
+            node: 22.19.0
+            confluent-kafka-version: 7.8.0
+            os: ubuntu-24.04
+          - label: Node.js 22.19.0 - Kafka 3.9.0
+            node: 22.19.0
+            confluent-kafka-version: 7.9.0
+            os: ubuntu-24.04
+          - label: Node.js 22.19.0 - Kafka 4.0.0
+            node: 22.19.0
+            confluent-kafka-version: 8.0.0
+            os: ubuntu-24.04
+
+          - label: Node.js 24.8.0 - Kafka 3.5.0
+            node: 24.8.0
+            confluent-kafka-version: 7.5.0
+            os: ubuntu-24.04
+          - label: Node.js 24.8.0 - Kafka 3.6.0
+            node: 24.8.0
+            confluent-kafka-version: 7.6.0
+            os: ubuntu-24.04
+          - label: Node.js 24.8.0 - Kafka 3.7.0
+            node: 24.8.0
+            confluent-kafka-version: 7.7.0
+            os: ubuntu-24.04
+          - label: Node.js 24.8.0 - Kafka 3.8.0
+            node: 24.8.0
+            confluent-kafka-version: 7.8.0
+            os: ubuntu-24.04
+          - label: Node.js 24.8.0 - Kafka 3.9.0
+            node: 24.8.0
+            confluent-kafka-version: 7.9.0
+            os: ubuntu-24.04
+          - label: Node.js 24.8.0 - Kafka 4.0.0
+            node: 24.8.0
+            confluent-kafka-version: 8.0.0
+            os: ubuntu-24.04
```

Comment:
Why did you do it in this way? I generically prefer ranges with exclusion

Reply:
Are you referring to Ubuntu or Node? Anyway I'll drop minor and patch from Node version.